kernel/src/rtc.rs

```diff
@@ -0,0 +1,121 @@
+use interrupts;
+use x86_util::{inb, outb};
+
+//Rate controls the interrupt frequency
+//Calculate with `frequency =  32768 >> (rate-1);`
+//Rate must be between 3 and 15 inclusice
+const RATE: u8 = 6;
+
+#[derive(Copy, Clone)]
+pub struct RtcTime {
+    pub seconds: u8,
+    pub minutes: u8,
+    pub hours: u8,
+    pub day_of_week: u8,
+    pub date_of_month: u8,
+    pub month: u8,
+    pub year: u8,
+    pub count: u64,
+}
+
+static TIME: spin::Mutex<RtcTime> = spin::Mutex::new(RtcTime {
```

Comment:
For conciseness, you can derive `Default` on `RtcTime` and just initialize the mutex with `Default::default()`